pkg/collector/scheduler.go

```diff
@@ -0,0 +1,207 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package collector
+
+import (
+	"expvar"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+	"github.com/DataDog/datadog-agent/pkg/collector/loaders"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var (
+	schedulerErrs  *expvar.Map
+	errorStats     = newCollectorErrors()
+	checkScheduler *CheckScheduler
```

Comment:
I'd be happy to re-discuss with you IRL what forces us to keep this around at the package level

Reply:
We re-discussed this and agreed this is the best solution right now so that the `check` command can still pull a specific `check.Check`s and run it directly.